### PR TITLE
improve matcher perf.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

File: test/Microsoft.DocAsCode.MarkdownLite.Tests/PerfTest.cs [hunk 1]
```diff
@@ -33,1116 +32,6 @@ public void TestPerf()
             GC.Collect();
         }
 
-        [Fact]
-        [Trait("Related", "Markdown")]
-        [Trait("Related", "Perf")]
```
Comment (ansyral) [inline test/Microsoft.DocAsCode.MarkdownLite.Tests/PerfTest.cs]: why remove?

**Comment by vicancy** [comment]
>     public class PerfTest

How about moving perf test out to a benchmark project?

---
Refers to: test/Microsoft.DocAsCode.MarkdownLite.Tests/PerfTest.cs:17 in f2fb530. [](commit_id = f2fb530a95e80f4866f647e6b51f9b70a28b7401, deletion_comment = True)

File: test/Microsoft.DocAsCode.MarkdownLite.Tests/PerfTest.cs (hunk 1 above)
Comment (vwxyzh) [inline test/Microsoft.DocAsCode.MarkdownLite.Tests/PerfTest.cs]: no use.

**Comment by vwxyzh** [comment]
@vicancy I will do it later.

**Pull request merged**